scripts/infra/local-setup.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env -S bash -e
+# SPDX-License-Identifier: Apache-2.0
+
+rh__ensure-aws-system-pkgs() {
+    # shellcheck source=/dev/null
+    source /etc/os-release
+    if [[ "$ID" == "fedora" ]] || [[ "$ID_LIKE" =~ "fedora" ]]; then
+        PKG_LIST=(git gcc python-devel python-virtualenv krb5-devel openldap-devel aws)
```

Comment:
These packages:

gcc python-devel python-virtualenv krb5-devel openldap-devel

Wouldn't be needed by everyone.  It depends on how you are authenticating to aws.

Reply:
Tbh I'm not sure where to go from here - this is really the meat of the PR - any suggestions?

Reply:
It hit me too when I made the comment.  You could have a separate command for ensure aws sso auth pkgs.  But the comment would need to be pretty wordy to explain when/why you would need it.  We could easily have other use cases for this script to setup a local instructlab environment.  Maybe we need to settle on the second one before the commit will be worthwhile though.

Reply:
I had this thought - we install the base packages (just `git` and `aws) with a flag `-s` to install packages required for SSO/SAML authentication. I'll push up an implementation that we can discuss further on.